### PR TITLE
Static reply app

### PR DESCRIPTION
**Pull request by jerith**
It would be useful to have a Vumi Go app that replies to all incoming message with a preconfigured reply. Also very straightforward to implement.

### CODE REVIEW

File: go/apps/static_reply/vumi_app.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    Application that replies to incoming messages with a configured response.
+    """
+    worker_name = 'static_reply_application'
+    CONFIG_CLASS = StaticReplyConfig
+
+    @inlineCallbacks
+    def consume_user_message(self, message):
+        config = yield self.get_message_config(message)
+        if config.reply_text:
+            yield self.reply_to(message, config.reply_text)
```
Comment (hodgestar) [inline go/apps/static_reply/vumi_app.py]: Setting this up on USSD with a suitable message could be amusing. :)

File: go/conversation/view_definition.py [hunk 2]
```diff
@@ -328,7 +328,10 @@ def process_forms(self, request, conversation):
             # Is this a good idea?
             if not edit_form.is_valid():
                 return self._render_forms(request, conversation, edit_forms)
-            config[key] = self.process_form(edit_form)
+            if key is None:
+                config = self.process_form(edit_form)
+            else:
+                config[key] = self.process_form(edit_form)
```
Comment (hodgestar) [inline go/conversation/view_definition.py:334]: Useful.

**Comment by hodgestar** [comment]
:+1:

File: go/conversation/view_definition.py (hunk 2 above)
Comment (jerith) [inline go/conversation/view_definition.py:334]: That was always supposed to be like that (and it works that way when building the forms from config data), but nothing actually used it and this side was broken.

File: go/apps/static_reply/vumi_app.py (hunk 1 above)
Comment (jerith) [inline go/apps/static_reply/vumi_app.py]: Maybe we should explicitly end the session here?

Comment (hodgestar) [inline go/apps/static_reply/vumi_app.py]: Explicitly ending the session sounds sensible.

**Pull request merged**